pom.xml

```diff
@@ -37,8 +37,10 @@
         <dependency>
             <groupId>org.assertj</groupId>
             <artifactId>assertj-core</artifactId>
-            <version>${assertj.version}</version>
+            <version>3.21.0</version>
```

Comment:
assertj version we are using is `3.23.1` why we need to change that?

Reply:
`assertThat` was already deprecated in Junit assertion. And they have removed it in higher version. you can use other methods to do the same. [Junit Assertions Docs](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html)

but if you require to use the assertThat it is available on Assertj. You need to import
`import static org.assertj.core.api.Assertions.assertThat;` to use.

Do not downgrade the `assertj.version`

Reply:
I had some problems with this package right after opened the project, i tried to change some configurations to make it to be compiled, it resolved when I changed the version to 3.21.0.
I'm still not sure how to resolve this issue in a better way, trying to resolve it right now. 
![Screenshot from 2023-07-09 22-53-25](https://github.com/TheAlgorithms/Java/assets/20107490/0c80d8f2-4bb0-4613-a00f-b7471e1940ac)
anyway, Its may not should be a part of this PR